communication/src/PlainUDPCommunication.cpp

```diff
@@ -251,25 +251,24 @@ class PlainUDPCommunication::PlainUdpImpl {
     const Addr *to = &nodes2addresses[destNode];
 
     ConcordAssert((to != NULL) && "The destination endpoint does not exist!");
-    ConcordAssert((messageLength > 0) && "The message length must be positive!");
-    ConcordAssert((message != NULL) && "No message provided!");
+    ConcordAssert((msg->size() > 0) && "The message length must be positive!");
 
     LOG_DEBUG(_logger,
-              " Sending " << messageLength << " bytes to " << destNode << " (" << inet_ntoa(to->sin_addr) << ":"
+              " Sending " << msg->size() << " bytes to " << destNode << " (" << inet_ntoa(to->sin_addr) << ":"
                           << ntohs(to->sin_port));
 
-    error = sendto(udpSockFd, message, messageLength, 0, (struct sockaddr *)to, sizeof(Addr));
+    error = sendto(udpSockFd, msg->data(), msg->size(), 0, (struct sockaddr *)to, sizeof(Addr));
 
     if (error < 0) {
       /** -1 return value means underlying socket error. */
       LOG_INFO(_logger, "Error while sending: " << concordUtils::errnoString(errno));
-    } else if (error < (int)messageLength) {
+    } else if (error < (int)msg->size()) {
```

Comment:
I know, this is not a new code, but would you mind to cast to `ssize_t`.

Reply:
Actually the size() returns size_t, but error is int. Without the cast the compilation fails:
`
/home/tdimitrov/concord-bft/communication/src/PlainUDPCommunication.cpp:265:22: error: comparison of integers of different signs: 'int' and 'std::vector::size_type' (aka 'unsigned long')
      [-Werror,-Wsign-compare]
    } else if (error < msg->size()) {
               ~~~~~ ^ ~~~~~~~~~~~
`